File: ch08/App_Code/VB/Listing8-9.vb
Imports System.Collections.Generic

Namespace Wrox.VB

    Public Class Customer
        Public Property CustomerID() As String
        Public Property CompanyName() As String
        Public Property ContactName() As String
        Public Property ContactTitle() As String
    End Class

    Public Class CustomerRepository

        Public Function [Select](ByVal customerID As String) As List(Of Customer)
            ' You would implement logic here to retreive
            ' Customer data based on the customerID parameter

            Dim _customers As New List(Of Customer)
            _customers.Add(New Customer() With {.CompanyName = "Acme", .ContactName = "Wiley Cyote", .ContactTitle = "President", .CustomerID = "ACMEC"})
            Return _customers
        End Function

        Public Sub Insert(ByVal c As Customer)
            ' Implement Insert logic
        End Sub

        Public Sub Update(ByVal c As Customer)
            ' Implement Update logic
        End Sub

        Public Sub Delete(ByVal c As Customer)
            ' Implement Delete logic
        End Sub

    End Class
End Namespace

File: ch08/App_Code/CS/Listing8-9.cs
using System.Collections.Generic;
namespace Wrox.CS
{
public class Customer
{
    public string CustomerID { get; set; }
    public string CompanyName { get; set; }
    public string ContactName { get; set; }
    public string ContactTitle { get; set; }
}

public class CustomerRepository
{
    public CustomerRepository()
    {
    }

    public List<Customer> Select(string customerId)
    {
        // Implement logic here to retrieve the Customer
        // data based on the methods customerId parameter
        List<Customer> _customers = new List<Customer>();
        _customers.Add(new Customer() { CompanyName = "Acme", ContactName = "Wiley Cyote", ContactTitle = "President", CustomerID = "ACMEC" });
        return _customers;
    }

    public void Insert(Customer c)
    {
        // Implement Insert logic
    }

    public void Update(Customer c)
    {
        // Implement Update logic
    }

    public void Delete(Customer c)
    {
        // Implement Delete logic
    }

}
}
